uhh.go

```diff
@@ -28,38 +29,67 @@ func contains(arr []string, str string) bool {
 }
 
 func (u *Uhh) Clone() error {
-	if !gitClone(u.config) {
+	if !gitClone(u.config.Repo(), u.config.LocalRepoPath()) {
 		return fmt.Errorf("Unable to clone")
 	}
 
 	return nil
 }
 
+func (u *Uhh) AddRepo(repoUrl string) error {
+    idx := len(u.config.vals.ReadRepos)
+    repoPath := path.Join(u.config.basePath, strconv.Itoa(idx))
+
+	if !gitClone(repoUrl, repoPath) {
+		return fmt.Errorf("Unable to clone")
+	}
+
+    u.config.vals.ReadRepos = append(u.config.vals.ReadRepos, strconv.Itoa(idx))
+
+    path, err := DefaultConfigPath()
+
+    if err != nil {
+        os.RemoveAll(repoPath)
+        return fmt.Errorf("%w", err)
+    }
+
+    err = u.config.Save(path)
+
+    if err != nil {
+        os.RemoveAll(repoPath)
+        return fmt.Errorf("%w", err)
+    }
+
+    return nil
+}
+
+func (u *Uhh) SetRepo() error {
+    return fmt.Errorf("Definitely not a JavaException here: NotImplemented")
+}
+
 func (u *Uhh) Sync() error {
-	if !gitCommitAdd(u.config) {
+	if !gitAdd(u.config) || !gitCommit(u.config) {
```

Comment:
We should probably update this method to stash changes, pull, apply changes, commit and push.

Then use this method in the `Add` method. `if u.config.SyncOnAdd { u.Sync() }`

Reply:
but sync on add should sync just the repo you are adding too?  Why would you want to pull from every repo?

Reply:
I was thinking it would simplify the users' mental model. `./uhh sync` -> sync all repos (pull read-only, update rw), `./uhh add --sync <cmd>` runs add, then syncs

both syncs mean the same thing.